File: 3396-ValidWord/3396-ValidWord.py
# Last updated: 31/07/2025, 21:05:20
class Solution:
    def isValid(self, word: str) -> bool:
        word=word.lower()
        d=c=v=0
        if len(word)>=3:
            for I in word:
                if I.isdigit():
                    d+=1
                elif I in "aeiou":
                    v+=1
                elif I in "bcdfghjklmnpqrstvwxyz":
                    c+=1
            if c>=1 and v>=1 and c+v+d==len(word):
                return  True
            else:
                return False
        else:
            return False                                



        

File: 3396-ValidWord/3396-ValidWord.java
// Last updated: 31/07/2025, 21:05:18
class Solution {
    public boolean isValid(String word) {
        if(word.length()<3)
        {
            return false;
        }
        word=word.toLowerCase();
        String vowels     = "aeiou";
        String consonants = "bcdfghjklmnpqrstvwxyz";
        int c=0,v=0,d=0;
        for(char ch:word.toCharArray())
        {
            if(Character.isDigit(ch))
            {
                d++;
            }
            else if(vowels.indexOf(ch)>=0)
            {
                v++;
            }
            else if(consonants.indexOf(ch)>=0)
            {
                c++;
            }
        }
        if(c>=1 && v>=1 && c+v+d==word.length())
        {
            return true;
        }
        return false;        
    }
}